docs(boilerplate): add AI agent skills documentation (#758)
commerce-docs/microsite-commerce-storefront: src/content/docs/boilerplate/ai-agent-skills.mdx

| **Drop-in developer** | Customizes drop-in components using containers, slots, events, and API functions |
| **Content modeler** | Designs block table structures that are easy for content authors to work with |
| **Tester** | Verifies implementations in a real browser and checks Core Web Vitals and accessibility |

</TableWrapper>

The skills work together. For every new task, your agent starts with the project manager skill to scope the work, consults the researcher skill before implementing, and delegates to the appropriate developer skill based on what needs to be built.

## Prerequisites

Before installing the skills, make sure you have the following:

- **Node.js 22 or later** — The Adobe I/O CLI requires Node 22+. Use <Link href="https://formulae.brew.sh/formula/nvm" text="nvm" /> to manage multiple Node versions.
- **An AEM Boilerplate Commerce project** — The skills are designed for projects based on the <Link href="https://github.com/hlxsites/aem-boilerplate-commerce" text="Commerce Boilerplate" />. See [Getting started](/boilerplate/getting-started/) if you need to create one first.

> - **Adobe I/O CLI** — Install or update with npm:

Reviewer: Are there requirements for authentication/login to use these tools? Like do you need to login via the aio cli?

Author: Addressed in [7f31de3](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/7f31de3ce79e1de2d824ab17fc17fd93875ce54d)

Essentially you don't need to login to install the skills but the MCP server does requires it.

  ```bash
  npm install -g @adobe/aio-cli
  ```

  See the <Link href="https://developer.adobe.com/app-builder/docs/guides/runtime_guides/tools/cli-install" text="Adobe I/O CLI installation guide" /> for more details.

- **Adobe I/O Commerce plugin** — Install the Commerce plugin for the aio CLI:

  ```bash
  aio plugins:install https://github.com/adobe-commerce/aio-cli-plugin-commerce
  ```

  See the <Link href="https://github.com/adobe-commerce/aio-cli-plugin-commerce" text="aio-cli-plugin-commerce repository" /> for additional details.

## Install the skills

Run the following command from the root of your boilerplate project:

```bash
aio commerce extensibility tools-setup
```

The command walks you through two prompts:

> <Steps>

Reviewer 2: What is the difference between the two? Can we please add some context similar to "Select a starter kit if you want to do XX"

Author: Addressed in [7f31de3](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/7f31de3ce79e1de2d824ab17fc17fd93875ce54d)

Added clarification that this is a series of prompts. One for selecting the correct starter kit to load the correct skill set and then a follow up for selecting the target agent, which determines where the skill files are copied into.

</Steps>

The command installs the `@adobe-commerce/commerce-extensibility-tools` package as a dev dependency, copies the skill files into your agent's skills directory, and configures MCP (Model Context Protocol) so your agent can access Commerce documentation search tools.

<Aside type="tip" title="Non-interactive install">
To install without prompts — for example, in a CI pipeline — pass all flags directly:

```bash
aio commerce extensibility tools-setup \
  --starter-kit aem-boilerplate-commerce \
  --agent Cursor \
  --package-manager npm
```

</Aside>

### Supported agents

<TableWrapper>

| Agent | Skills location |
|-------|-----------------|
| Cursor | `.cursor/skills/` |
| Claude Code | `.claude/skills/` |
| GitHub Copilot | `.github/skills/` |
| Windsurf | `.windsurf/skills/` |
| Gemini CLI | `.gemini/skills/` |
| OpenAI Codex | `.agents/skills/` |
| Cline | `.cline/skills/` |
| Kilo Code | `.kilocode/skills/` |
| Antigravity | `.agent/skills/` |
| Other | `./skills/` (project root) |

</TableWrapper>

After the install completes, restart your coding agent so it picks up the new skills and MCP configuration.

## How to use the skills

> Once installed, the skills are available to your agent automatically. You don't need to reference them by name in every prompt. Here is how the agent uses them:

Reviewer: How would a user confirm that the skills are properly configured/set up?

Author: Addressed in [7f31de3](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/7f31de3ce79e1de2d824ab17fc17fd93875ce54d)

There are different ways to check this and it depends on the agent/ide.

---
title: AI agent skills
description: Learn how to install and use AI agent skills for Adobe Commerce storefront development on Edge Delivery Services with the AEM Commerce Boilerplate.
sidebar:
  label: AI agent skills
  order: 8
---

import Aside from '@components/Aside.astro';
import TableWrapper from '@components/TableWrapper.astro';
import Link from '@components/Link.astro';
import { Steps } from '@astrojs/starlight/components';
import { Tabs, TabItem } from '@astrojs/starlight/components';

> AI agent skills give your coding agent deep, domain-specific knowledge about Adobe Commerce storefront development. When you install the **AEM Boilerplate Commerce** skill set, your agent understands the boilerplate's architecture, drop-in component patterns, block conventions, and best practices — so you spend less time explaining context and more time building.

Reviewer 3: Is this a "skill set" or a "starter kit"? if we are not referring to it as a "starter kit", then maybe the CLI should be changed to a broader term.

Author: Addressed in [5bace23](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/5bace23fa1eefb31235014f05c903a1a2dedba96)

Added a clarification on what a starter kit is. Changes to the CLI will need to be done in another story.

| **Researcher** | Looks up drop-in component APIs, slot names, event payloads, and TypeScript definitions before implementing |
| **Block developer** | Builds and customizes Edge Delivery Services blocks using correct DOM patterns and CSS scoping |
| **Drop-in developer** | Customizes drop-in components using containers, slots, events, and API functions |
| **Content modeler** | Designs block table structures that are easy for content authors to work with |
| **Tester** | Verifies implementations in a real browser and checks Core Web Vitals and accessibility |

</TableWrapper>

The skills work together. For every new task, your agent starts with the project manager skill to scope the work, consults the researcher skill before implementing, and delegates to the appropriate developer skill based on what needs to be built.

## Prerequisites

Before installing the skills, make sure you have the following:

- **Node.js 22 or later** — The Adobe I/O CLI requires Node 22+. Use <Link href="https://formulae.brew.sh/formula/nvm" text="nvm" /> to manage multiple Node versions.

> - **An AEM Boilerplate Commerce project** — The skills are designed for projects based on the <Link href="https://github.com/hlxsites/aem-boilerplate-commerce" text="Commerce Boilerplate" />. See [Getting started](/boilerplate/getting-started/) if you need to create one first.

Reviewer 3: ```suggestion
- **An AEM Boilerplate Commerce project** — The skills are designed for projects based on the <Link href="https://github.com/hlxsites/aem-boilerplate-commerce" text="Commerce Boilerplate" />. If you have not created a boilerplate storefront, see [Getting started](/boilerplate/getting-started/) to create one.
```

- **Adobe I/O CLI** — Install or update with npm:

  ```bash
  npm install -g @adobe/aio-cli
  ```

  See the <Link href="https://developer.adobe.com/app-builder/docs/guides/runtime_guides/tools/cli-install" text="Adobe I/O CLI installation guide" /> for more details.

- **Adobe I/O Commerce plugin** — Install the Commerce plugin for the aio CLI:

  ```bash
  aio plugins:install https://github.com/adobe-commerce/aio-cli-plugin-commerce
  ```

  See the <Link href="https://github.com/adobe-commerce/aio-cli-plugin-commerce" text="aio-cli-plugin-commerce repository" /> for additional details.

<Aside type="note" title="Authentication and entitlements">

> You do not need to log in via the Adobe I/O CLI (IMS), or have any specific org or entitlement, to install the skills. The MCP server that the setup configures — and that the researcher skill uses for documentation search — can use IMS when available. If that service is not available to your agent, the researcher skill falls back to web search instead.

Reviewer 3: You might add that if they want to login they should use `aio login`. and i would reframe it as stating the limits like "without authentication, X doesn't work."

Author: Done [5bace23](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/5bace23fa1eefb31235014f05c903a1a2dedba96)

<Aside type="tip" title="Non-interactive install">
To install without prompts — for example, in a CI pipeline — pass all flags directly:

```bash
aio commerce extensibility tools-setup \
  --starter-kit aem-boilerplate-commerce \
  --agent Cursor \
  --package-manager npm
```

</Aside>

### Supported agents

<TableWrapper>

| Agent | Skills location |
|-------|-----------------|
| Cursor | `.cursor/skills/` |
| Claude Code | `.claude/skills/` |
| GitHub Copilot | `.github/skills/` |
| Windsurf | `.windsurf/skills/` |
| Gemini CLI | `.gemini/skills/` |
| OpenAI Codex | `.agents/skills/` |
| Cline | `.cline/skills/` |
| Kilo Code | `.kilocode/skills/` |
| Antigravity | `.agent/skills/` |
| Other | `./skills/` (project root) |

</TableWrapper>

![CLI output after a successful skills install: setup complete, skills copied, and next steps](@images/boilerplate/aio-cli-commerce-skills-result.png)

After the install completes, restart your coding agent so it picks up the new skills and MCP configuration.

> Different agents and IDEs expose skills in different ways, so there is no single way to confirm that the skills are installed correctly. Check your agent's or IDE's documentation for how to verify that it has loaded your project's skills. For example, in Cursor you can open the configuration or settings area to see the list of skills it has detected; other agents use different UIs or indicators.

Reviewer 3: maybe move the table below this section

Author: Done [5bace23](https://github.com/commerce-docs/microsite-commerce-storefront/pull/758/commits/5bace23fa1eefb31235014f05c903a1a2dedba96)
